### PR TITLE
Add test for @code_typed and @code_lowered

### PR DESCRIPTION
**Pull request by ranjanan**
Check if @code_typed and @code_lowered return LambdaInfo
instead of Array.

Test for PR #17636.

### CODE REVIEW

**Comment by tkelman** [comment]
best to also test for the case where `gen_call_with_extracted_types` returns multiple results, if you can come up with a case where it does that

File: test/inference.jl [hunk 1]
```diff
@@ -276,4 +276,11 @@ let
     b = @code_lowered 1 + 1
     @test isa(a, LambdaInfo)
     @test isa(b, LambdaInfo)
+
+    Base.view(A::AbstractArray, i::Real) = view(A, :, i)
```
Comment (tkelman) [inline test/inference.jl]: could we make up a new local function here as opposed to extending an important base function for this test?

Comment (ranjanan) [inline test/inference.jl]: Ah yes. I can.

**Comment by JeffBezanson** [comment]
It doesn't seem possible for `gen_call_with_extracted_types` to return multiple results, since it uses concrete types. There has to be one most specific match, or else an ambiguity error.

**Comment by tkelman** [comment]
Ah, my mistake. maybe this should change (later, not here) to return a Nullable then, rather than LambdaInfo-or-empty-array?

These tests might be better in test/reflection.jl.

**Comment by JeffBezanson** [comment]
Yes, it should possibly return LambdaInfo-or-error. Nullable would be fine, except still has the frustrating printing:

```
julia> Nullable(@code_typed 1+0)
Nullable{LambdaInfo}(LambdaInfo for +(::Int64, ::Int64))
```

**Comment by ranjanan** [comment]
Looks like a Travis build timed out.

**Pull request merged**